src/gt4py/next/program_processors/runners/dace_fieldview/gtir_to_tasklet.py

```diff
@@ -0,0 +1,207 @@
+# GT4Py - GridTools Framework
+#
+# Copyright (c) 2014-2023, ETH Zurich
+# All rights reserved.
+#
+# This file is part of the GT4Py project and the GridTools framework.
+# GT4Py is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or any later
+# version. See the LICENSE.txt file at the top-level directory of this
+# distribution for a copy of the license or check <https://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+
+from dataclasses import dataclass
+from typing import TypeAlias
+
+import dace
+import dace.subsets as sbs
+
+from gt4py import eve
+from gt4py.next.common import Connectivity, Dimension
+from gt4py.next.iterator import ir as itir
+from gt4py.next.iterator.ir_utils import common_pattern_matcher as cpm
+from gt4py.next.program_processors.runners.dace_fieldview.gtir_python_codegen import (
+    MATH_BUILTINS_MAPPING,
+)
+from gt4py.next.program_processors.runners.dace_fieldview.utility import as_dace_type, unique_name
+
+
+@dataclass(frozen=True)
+class MemletExpr:
+    """Scalar or array data access thorugh a memlet."""
+
+    source: dace.nodes.AccessNode
+    subset: sbs.Indices | sbs.Range
+
+
+@dataclass(frozen=True)
+class SymbolExpr:
+    """Any symbolic expression that is constant in the context of current SDFG."""
+
+    value: dace.symbolic.SymExpr
+    dtype: dace.typeclass
+
+
+@dataclass(frozen=True)
+class TaskletExpr:
+    """Result of the computation provided by a tasklet node."""
+
+    node: dace.nodes.Tasklet
+    connector: str
+
+
+@dataclass(frozen=True)
+class IteratorExpr:
+    """Iterator for field access to be consumed by `deref` or `shift` builtin functions."""
+
+    field: dace.nodes.AccessNode
+    dimensions: list[str]
+    offset: list[dace.symbolic.SymExpr]
```

Comment:
The attribute `offset` can be removed. I will remove it in next update and implement cartesian shift by adding symbolic expressions in `index` attribute.